packages/next/src/build/webpack/loaders/next-metadata-image-loader.ts

```diff
@@ -172,7 +172,7 @@ async function nextMetadataImageLoader(
     const imageData = ${JSON.stringify(imageData)}
     const imageUrl = fillMetadataSegment(${JSON.stringify(
       pathnamePrefix
-    )}, props.params, ${JSON.stringify(pageSegment)})
+    )}, await props.params, ${JSON.stringify(pageSegment)})
```

Comment:
The function needs to be `async` now. Also make sure to check that each callsite is now also awaited.